### ISSUE TITLE
Add ability to specify a proxy host and port in the UI

### ISSUE THREAD

**Issue by matthewhorridge** [protegeproject/protege#530]
From the mailing list:

>I use Protege 5.0.0 under Windows and have to configure a proxy to be able to install plugins (I need a plugin to import from XML-file).
In the documentation it says
 
>“Edit the Protege.lax file with a text editor. The required information can be included at the bottom of the lax file:”
 
>But I can’t find the Protege.lax file.

**Comment by rwynne** [comment]
Probably requires more input for the SSL part.

HTTPS or HTTP via an Apache "proxy"?

**Comment by tudorache** [comment]
This is a JVM setting.
Setting the java proxy args in run.sh and run.bar should work, something like:
 ```-Dhttp.proxyHost=10.0.0.100 -Dhttp.proxyPort=8800```

See more here:
http://docs.oracle.com/javase/6/docs/technotes/guides/net/proxies.html 
and here:
http://stackoverflow.com/questions/120797/how-do-i-set-the-proxy-to-be-used-by-the-jvm

**Comment by matthewhorridge** [comment]
The question is whether there should be a UI in Protege to set these settings.

**Comment by tudorache** [comment]
I don't think so, but we should document this (maybe FAQ?), although this is not a Protege setting, but rather a generic Java one.

**Comment by matthewhorridge** [comment]
It's app specific though....that's my thought.

**Comment by felixbur** [comment]
thing is, although I added
jre\bin\java  -Dhttp.proxyHost=212.201.104.11 -Dhttp.proxyPort=8080 
to run.bat
and Protege.l4j.ini
(which worked for other programs needing internet access)

I still get

--- Searching for plugins ---
Checking https://raw.githubusercontent.com/protegeproject/autoupdate/master/upda
te-info/5.0.0/plugins.repository
    Processing https://raw.githubusercontent.com/protegeproject/autoupdate/maste
r/update-info/5.0.0/plugins.repository as a plugin registry
        Cannot read plugin registry at https://raw.githubusercontent.com/protege
project/autoupdate/master/update-info/5.0.0/plugins.repository.  Reason: Connect
ion timed out: connect

**Comment by matthewhorridge** [comment]
Hi Felix,

Is it just the auto-update that does not work?  Are you able to load ontologies directly from URLs?

**Comment by felixbur** [comment]
Now that you mention it: if I go to "open from url" and click one of the urls already provided, e.g. "vines" then this ontology gets loaded. 
Trying to check for plugins still gives the connection error.
There's also a registry problem, but then I don't have admin rights on the machine:

Sep 21, 2016 9:48:50 AM java.util.prefs.WindowsPreferences <init>
WARNING: Could not open/create prefs root node Software\JavaSoft\Prefs at root 0
x80000002. Windows RegCreateKeyEx(...) returned error code 5.
Creating and setting up empty (default) editor kit
OWL API Version: 4.2.5.20160517-0735

**Comment by Evertude** [comment]
Auto update uses probably https protocol because when I added -Dhttps.proxyHost=host -Dhttps.proxyPort=port to the .bat it worked. Please try it out

**Comment by felixbur** [comment]
THanks, you were right, adding https did the trick

**Comment by headchant** [comment]
Is it possible to authenticate with a proxy using username and password? I tried to use the -Dhttps.proxyUser and -Dhttps.proxyPassword but I still get "HTTP/1.0 407 Proxy Authentication Required". I tried setting the jdk.http.auth.tunneling.disabledSchemes="" to empty but that did not work either. Any hints? Does protege even implement any kind of authentication?

**Comment by Evertude** [comment]
https://stackoverflow.com/questions/1626549/authenticated-http-proxy-with-java

-Dhttp.proxyUser=someUserName
-Dhttp.proxyPassword=somePassword
-Dhttps.proxyUser=someUserName
-Dhttps.proxyPassword=somePassword

**Comment by Evertude** [comment]
Here you can even find an implementation for the proxy, should help with solving the feature:
https://stackoverflow.com/questions/1626549/authenticated-http-proxy-with-java

-Dhttp.proxyUser=someUserName
-Dhttp.proxyPassword=somePassword
-Dhttps.proxyUser=someUserName
-Dhttps.proxyPassword=somePassword

**Comment by visraj1** [comment]
Hi,
Tried updating the .bat file with -Dhttps as well. But still the auto update results in connection timed out?

**Comment by matthewhorridge** [comment]
Slated fo 6.0.0
